src/main/java/com/hivemq/embedded/internal/EmbeddedExtensionBuilderImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019-present HiveMQ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hivemq.embedded.internal;
+
+import com.google.common.base.Preconditions;
+import com.hivemq.embedded.EmbeddedExtension;
+import com.hivemq.embedded.EmbeddedExtensionBuilder;
+import com.hivemq.extension.sdk.api.ExtensionMain;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.extension.sdk.api.annotations.Nullable;
+
+public class EmbeddedExtensionBuilderImpl implements EmbeddedExtensionBuilder {
+
+    private @Nullable String id;
+    private @Nullable String name;
+    private @Nullable String version;
+    private @Nullable String author;
+    private int priority = 0;
+    private int startPriority = 1000;
```

Comment:
the priorities should use constants, maybe also share the same constants as the other extension code